File: java/android_level_4/app/src/main/java/bliepy/com/android_level_4/AddActivity.java
package bliepy.com.android_level_4;

import android.app.ActionBar;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.TextView;

import bliepy.com.android_level_4.database.GameBacklogModel;

public class AddActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                addItem ();
                Intent intent = new Intent(getApplicationContext(), OverviewActivity.class);
                startActivity(intent);
            }
        });
    }


    public void addItem (){
        TextView titleView = findViewById(R.id.overviewTitle);
        TextView platformView = findViewById(R.id.textView1);
        TextView noteView = findViewById(R.id.textView2);
        TextView playView = findViewById(R.id.textView3);

        String title = titleView.getText().toString();
        String platform = platformView.getText().toString();
        String note = noteView.getText().toString();
        String play = playView.getText().toString();

        GameBacklogModel model = new GameBacklogModel();
        model.setTitle(title);
        model.setPlatform(platform);
        model.setNotes(note);
        model.setPlatform(play);
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: kotlin/level_5_task_2/app/src/main/java/com/bliepy/gamebacklog/AddActivity.kt
package com.bliepy.gamebacklog

import android.content.Intent
import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProviders
import com.bliepy.gamebacklog.model.GameEntity
import com.bliepy.gamebacklog.viewmodel.GameViewModel
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_add.*
import java.lang.Exception
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class AddActivity : AppCompatActivity() {

    private var gameViewModel: GameViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add)
        setSupportActionBar(toolbar)
        gameViewModel = ViewModelProviders.of(this).get(GameViewModel::class.java)
        saveAction()
    }

    fun saveAction (){
        fab.setOnClickListener {

            val gameTitle: EditText = findViewById(R.id.AddTitle)
            val gamePlatform: EditText = findViewById(R.id.AddPlatform)
            val gameDateDay : EditText= findViewById(R.id.AddDays)
            val gameDateMonth: EditText = findViewById(R.id.AddMonths)
            val gameDateYear: EditText = findViewById(R.id.AddYears)

            val title = gameTitle?.text.toString()
            val platform = gamePlatform?.text.toString()
            val days = gameDateDay.text.toString()
            val months = gameDateMonth.text.toString()
            val years = gameDateYear.text.toString()

            try {

                val date: LocalDate = LocalDate.parse("$years-$months-$days", DateTimeFormatter.ISO_DATE)
                gameViewModel?.createGame(
                    GameEntity(
                        title = title,
                        platform = platform,
                        date = date.toEpochDay()
                    )
                )
                startActivity(Intent(this, MainActivity::class.java))
            }catch (e: Exception){
                Snackbar.make(it, "Invalid date", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
            }
        }
    }
}
